### ISSUE TITLE
unstable training

### ISSUE THREAD

**Issue by zrx2000** [Hans1984/SelfTMO#2]
First, I’d like to thank you for kindly sharing the source code of this great work. However, my training seems to be very unstable. Sometimes I get an image that is basically normal, but sometimes I get an image with many defect pixels and artifacts. And as the iteration increases, the loss tends to become nan and then I get a black image. Some results I got with the same image are shown below. Could you please suggest potential strategies to make the training process more stable? Thank you!

![Image](https://github.com/user-attachments/assets/600306d6-598e-4c48-9f8c-abfba69f8238)
![Image](https://github.com/user-attachments/assets/0c27dee1-bb93-4112-9553-b38ff2ba134b)
![Image](https://github.com/user-attachments/assets/19a5cd67-76d2-48c2-8b01-ecce4e3ad5c4)
![Image](https://github.com/user-attachments/assets/bee4f816-1591-42ff-9264-53d48f73275a)
![Image](https://github.com/user-attachments/assets/c0ce26ee-5f2d-41c3-b742-ba940f8146d3)

**Comment by Hans1984** [comment]
Hello, thank you for your interest. 
I haven't encountered your issue before. This TensorFlow version is quite old. However, I think you could try setting a fixed training seed, as mentioned here: https://github.com/Hans1984/SelfTMO/issues/1

**Comment by zrx2000** [comment]
Thanks for your reply! I'll give it a try.
